frontend/src/pages/DeptPage.jsx

```diff
@@ -47,9 +48,28 @@ export default function DeptPage({ college, dept, setCourse }) {
   const [classes, error2] = useApi(`/${college}/depts/${dept}/classes`)
   const colleged = campus.find((cmp) => cmp.id === college)
 
-  const cards = courses
-    ? courses.map(({ dept, course, title, classes }) => (
-      <DeptCard
+  const [query, setQuery] = useState('')
+  const [filteredCourses, setFilteredCourses] = useState([])
+  const [filteredClasses, setFilteredClasses] = useState([])
+
+  useEffect(() => {
+    if (courses && classes) {
+      const filteredCourses = matchSorter(courses, query, {
+        keys: [
+          item => item.dept + ' ' + item.course,
+          {threshold: matchSorter.rankings.EQUAL, key: 'course'},
+          {minRanking: matchSorter.rankings.MATCHES, key: 'title'}
+        ]
+      })
+      setFilteredCourses(filteredCourses)
+      
+      const filteredCRNS = new Set(filteredCourses.map(course => course.classes).flat())
+      setFilteredClasses(classes.filter(c => filteredCRNS.has(c.CRN)))
+    }
+  }, [courses, classes, query])
+
+  const cards = (filteredCourses && filteredCourses.length) || courses
+    ? ((filteredCourses.length && filteredCourses) || courses).map(({ dept, course, title, classes }) => <DeptCard
```

Comment:
Maybe make a separate var to make this cleaner 😄

Reply:
okay will do this.